app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [Person(person["name"], person["age"]) for person in people]
+
+    for person in people:
+        person_instance = Person.people[person["name"]]
+        if "wife" in person and person["wife"] is not None:
+            person_instance.wife = Person.people[person["wife"]]
+        elif "husband" in person and person["husband"] is not None:
+            person_instance.husband = Person.people[person["husband"]]
```

Comment:
The code assumes that if a person has a 'wife' or 'husband', they are already present in the `Person.people` dictionary. Ensure that all persons, including spouses, are included in the initial `people` list to avoid potential `KeyError` exceptions.